qt/widgets/mplcpp/test/PlotTest.h

```diff
@@ -0,0 +1,158 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MPLCPP_PLOTTEST_H
+#define MPLCPP_PLOTTEST_H
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidQtWidgets/MplCpp/Plot.h"
+
+using namespace MantidQt::Widgets::Common;
+using namespace MantidQt::Widgets::MplCpp;
+
+namespace {
+void setMatplotlibBackend() {
+  PyObject *functionsString = Py_BuildValue("s", "matplotlib");
```

Comment:
Can we avoid using the raw Python api and use the C++ object api like suggested in `plot`. If we don't the reference counting gets very messy.